test/CollectionTest.cc

```diff
@@ -915,3 +936,85 @@ TEST_CASE_METHOD(CollectionTest, "Delete Scope and Close Database then Use Scope
     
     CBLScope_Release(scope);
 }
+
+#pragma mark - LISTENERS:
+
+TEST_CASE_METHOD(CollectionTest, "Collection notifications") {
+    // Add a listener:
+    defaultListenerCalls = fooListenerCalls = 0;
+    auto token = CBLCollection_AddChangeListener(defaultCollection, defaultListener, this);
+    auto docToken = CBLCollection_AddDocumentChangeListener(defaultCollection, "foo"_sl, fooListener, this);
+
+    // Create a doc, check that the listener was called:
+    createDocWithPair(defaultCollection, "foo", "greeting", "Howdy!");
+    CHECK(defaultListenerCalls == 1);
+    CHECK(fooListenerCalls == 1);
+
+    CBLListener_Remove(token);
+    CBLListener_Remove(docToken);
+
+    // After being removed, the listener should not be called:
+    defaultListenerCalls = fooListenerCalls = 0;
+    createDocWithPair(defaultCollection, "bar", "greeting", "yo.");
+    CHECK(defaultListenerCalls == 0);
+    CHECK(fooListenerCalls == 0);
+}
+
+TEST_CASE_METHOD(CollectionTest, "Remove Collection Listener after releasing collection") {
+    // Add a listener:
+    defaultListenerCalls = fooListenerCalls = 0;
+    auto token = CBLCollection_AddChangeListener(defaultCollection, defaultListener, this);
+    auto docToken = CBLCollection_AddDocumentChangeListener(defaultCollection, "foo"_sl, fooListener, this);
+
+    // Create a doc, check that the listener was called:
+    createDocWithPair(defaultCollection, "foo", "greeting", "Howdy!");
+    CHECK(defaultListenerCalls == 1);
+    CHECK(fooListenerCalls == 1);
+
+    CBLCollection_Release(defaultCollection);
+    defaultCollection = nullptr;
```

Comment:
Is it required to release default collection here?

Reply:
the idea behind the test is to check if listeners are successfully removed if the collection is release beforehand and as those are looking at default, yes. 

default is release at end of every test as well.. so, although the test passes, I don't know if a "double" release is a good of way of doing it. Another way would be to use a new collection, but that would need another listener as all four, at the moment, are for default.

Reply:
Oh, I see.